cmd/controller/run.go

```diff
@@ -165,6 +166,9 @@ func Run(opts Options, runLog logr.Logger) error {
 	refTracker := reftracker.NewAppRefTracker()
 	updateStatusTracker := reftracker.NewAppUpdateStatus()
 
+	// initialize deploy factory once - the deploy factory contains a cache which should be only setup once.
```

Comment:
maybe worth saying "cache of _"  (service accounts? right?) to assist future readers in their journey

Reply:
Yeh, we can make it more specific.